python/GafferUI/BoxUI.py

```diff
@@ -241,6 +241,12 @@ def __appendPlugPromotionMenuItems( menuDefinition, plug, readOnly = False ) :
 			"active" : not readOnly,
 		} )
 
+		if isinstance( plug.parent(), Gaffer.ArrayPlug ) and not box.plugIsPromoted( plug.parent() ) :
```

Comment:
It would be better for `box.canPromotePlug()` here, rather than `box.plugIsPromoted()`.